ansible_ai_connect/ai/api/views.py

```diff
@@ -1012,3 +1016,77 @@ def write_to_segment(
                 },
             )
         send_segment_event(event, "codegenPlaybook", user)
+
+
+class Chat(APIView):
+    """
+    Send a message to the backend chatbot service and get a reply.
+    """
+
+    permission_classes = [
+        permissions.IsAuthenticated,
+        IsAuthenticatedOrTokenHasScope,
+    ]
+    required_scopes = ["read", "write"]
+
+    CHAT_API_NOT_ENABLED = "Chat API is not enabled."
+    INVALID_CHAT_REQUEST = "An invalid chat request was received."
+    INVALID_CHAT_RESPONSE = "An invalid chat response was received."
+
+    def __init__(self):
+        self.chat_enabled = (
+            settings.CHATBOT_URL
+            and settings.CHATBOT_DEFAULT_MODEL
+            and settings.CHATBOT_DEFAULT_PROVIDER
+        )
+
+    @extend_schema(
+        request=ChatRequestSerializer,
+        responses={
+            200: ChatResponseSerializer,
+            400: OpenApiResponse(description="Bad Request"),
+        },
+        summary="Chat request",
+    )
+    def post(self, request) -> Response:
+        if not self.chat_enabled:
+            logger.warning(Chat.CHAT_API_NOT_ENABLED)
+            raise Exception(Chat.CHAT_API_NOT_ENABLED)
+
+        headers = {"Content-Type": "application/json"}
+        request_serializer = ChatRequestSerializer(data=request.data)
+
+        try:
+            if not request_serializer.is_valid():
+                logger.error(Chat.INVALID_CHAT_REQUEST)
+                raise Exception(Chat.INVALID_CHAT_REQUEST)
```

Comment:
It'd be nice to raise custom `Exception`'s with the `code` (and `message`) for easier handling in VSCode (should chat get to VSCode.. I think it might following discussions @jameswnl had with IBM). If the consumer is not VSCode custom exceptions do no harm.

Reply:
Yes, I will be updating the code in that way. Thanks.